iroha/src/crypto.rs

```diff
@@ -16,24 +17,57 @@ use ursa::{
 
 /// Represents hash of Iroha entities like `Block` or `Transaction.
 pub type Hash = [u8; 32];
+
+/// Pair of Public and Private keys.
+#[derive(Clone)]
+pub struct KeyPair {
+    /// Public Key.
+    pub public_key: PublicKey,
+    /// Private Key.
+    pub private_key: PrivateKey,
+}
+
 /// Public Key used in signatures.
-pub type PublicKey = [u8; 32];
+#[derive(Encode, Decode, PartialEq, Eq, PartialOrd, Debug, Clone, Hash, Default)]
+pub struct PublicKey {
+    origin: [u8; 32],
+}
+
+impl Ord for PublicKey {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.origin.cmp(&other.origin)
+    }
+}
+
+impl From<Vec<u8>> for PublicKey {
+    fn from(vector: Vec<u8>) -> Self {
+        let mut origin = [0; 32];
+        origin.copy_from_slice(&vector);
```

Comment:
It's dangerous to use this function because it can `panic!`. What about `TryFrom`?